### PR TITLE
RavenDB-12354 Allow to filter out Studio notifications

### PR DESCRIPTION
**Pull request by iftahbe**

### CODE REVIEW

File: src/Raven.Server/Config/Categories/StudioConfiguration.cs [hunk 1]
```diff
@@ -1,13 +1,78 @@
...
     {
-        [Description("The directory into which RavenDB will search the studio files, defaults to the base directory")]
+        [Description("The directory in which RavenDB will search the studio files, defaults to the base directory")]
         [DefaultValue(null)]
         [ConfigurationEntry("Studio.Path", ConfigurationEntryScope.ServerWideOnly)]
         public string Path { get; set; }
+        
+        [Description("Semicolon seperated list of notification names which will not be shown in the Studio. If not specified, all notifications are allowed. Example: \"SlowIO;Server_NewVersionAvailable;ClusterTopologyWarning\".")]
+        [DefaultValue(null)]
+        [ConfigurationEntry("Studio.Notifications.FilterOut", ConfigurationEntryScope.ServerWideOnly)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/StudioConfiguration.cs]: That isn't a Studio decisions, it belongs to notifications

File: src/Raven.Server/Config/Categories/StudioConfiguration.cs [hunk 2]
```diff
@@ -1,13 +1,78 @@
...
+        {
+            base.Initialize(settings, serverWideSettings, type, resourceName);
+
+            if (type != ResourceType.Server)
+                return;
+
+            ValidateFilteredOutNotificationNames();
+        }
+
+        private void ValidateFilteredOutNotificationNames()
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/StudioConfiguration.cs]: Let's not do that, we might change this and we don't want to break things if the user has a configuration for new version, frex.

File: src/Raven.Server/Config/Categories/StudioConfiguration.cs [hunk 3]
```diff
@@ -1,13 +1,78 @@
...
-        [Description("The directory into which RavenDB will search the studio files, defaults to the base directory")]
+        [Description("The directory in which RavenDB will search the studio files, defaults to the base directory")]
         [DefaultValue(null)]
         [ConfigurationEntry("Studio.Path", ConfigurationEntryScope.ServerWideOnly)]
         public string Path { get; set; }
+        
+        [Description("Semicolon seperated list of notification names which will not be shown in the Studio. If not specified, all notifications are allowed. Example: \"SlowIO;Server_NewVersionAvailable;ClusterTopologyWarning\".")]
+        [DefaultValue(null)]
+        [ConfigurationEntry("Studio.Notifications.FilterOut", ConfigurationEntryScope.ServerWideOnly)]
+        public string[] FilteredOutNotifications { get; set; }
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/StudioConfiguration.cs]: You are always comparing on this, easier if you will use `HashSet<string>` (case insensitive)

**Review by ayende — CHANGES_REQUESTED** [review]

**Review by ppekrol — CHANGES_REQUESTED** [review]
All of this should be moved to:
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Client/ServerWide/Operations/Configuration/ServerWideStudioConfiguration.cs
or
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Client/Documents/Operations/Configuration/StudioConfiguration.cs

it depends if we want to have it configurable server-wide only or database and server-wide.

After that we can add it to Studio.

**Comment by arekpalinski** [comment]
Please create a test

**Comment by iftahbe** [comment]
@ppekrol I don't follow why you want this. 
It is all handled server side, and the Studio is not aware of the filtered out notifications.

About the question whether it should also be per-database... I assumed that people will filter out mostly performance hints which should be server wide but if you think there are use-cases where people need this per database I can change that.

**Comment by ppekrol** [comment]
I've meant that we should move `public HashSet<string> FilteredOutNotifications { get; set; }` to `StudioConfiguration.cs` and let @ml054 add this to the UI, so users can configure it via Studio when Server is running.

Filtering should be done server-side as you do it now.

**Comment by ayende** [comment]
@ppekrol One of the reasons to do this is to allow the admin to shut down notifications.
This shouldn't be something that any user can edit.

I'm fine with having _additional_ items that are filtered, but the base line is fixed

**Comment by ppekrol** [comment]
@iftahbe add what I've wrote about to 4.2. For 4.1 this is enough.

**Pull request merged**